ledger-core/insolar/payload/validation.go

```diff
@@ -8,32 +8,200 @@ package payload
 import (
 	"github.com/insolar/assured-ledger/ledger-core/pulse"
 	"github.com/insolar/assured-ledger/ledger-core/reference"
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
 )
 
 type Validatable interface {
 	Validate(currPulse PulseNumber) error
 }
 
-func isTimePulseBefore(pn pulse.Number, before pulse.Number) bool {
-	return pn.IsTimePulse() && pn.IsBefore(before)
+type PulseValidator interface {
+	ValidateTime(pn pulse.Number, before pulse.Number) bool
+	Relation() string
+}
+
+type ExtendedPulseValidator interface {
```

Comment:
@bigbes If this interface named "extended" PulseValidator maybe we can inherit it?
Signature methods is matched
```
type ExtendedPulseValidator interface {
	PulseValidator
	GetSpecialValidator() PulseValidator
	ValidateSpecialOrTime(pn pulse.Number, before pulse.Number) bool
}
```

Reply:
fixed